### DISCUSSION TITLE
Custom DNS (Preview)

### DISCUSSION THREAD

**Discussion by Yavari** [polymind-inc/acmebot#488] [Show and tell]
I am going to implement a function for my custom DNS provider Loopia but are confused by the documentation at https://github.com/shibayan/keyvault-acmebot/wiki/DNS-Provider-Configuration#custom-dns-preview.

List Zones API:
Is it supposed to return the two name server IPs for the domain?

Upsert record: 
- PUT zones/{zoneId}/records/{recordName}
- Request body
    {
      "type": "TXT",
      "ttl": 60,
      "values": ["xxxxxx", "yyyyyy"]
    }

Is *zoneId* the domain and *recordName *the subdomain? 
What are the two values in that request body in that case? I thought a TXT record only can have one value?

**Comment by shibayan** [comment]
Documentation has been updated to include specific examples.

When the List zones API returns a name server, Acmebot automatically checks to see if the domain is pointing to the correct name server. However, this is an optional field.

The zone ID will be the id returned by the List Zone API and can be anything that identifies the domain. TXT records can have multiple values. This is required when issuing multiple domain certificates for `example.com` and `*.example.com`.

**Reply by Yavari** [reply]
Thank you. 

I have created an Azure Function for Loopia using custom DNS provider. Not sure if you want to link to external providers not maintained by you but you can find it here if you want to: https://github.com/Yavari/Acmebot.Provider.Loopia

**Reply by shibayan** [reply]
Your project has been added to the Wiki. Thank you!
https://github.com/shibayan/keyvault-acmebot/wiki/DNS-Provider-Configuration#related-projects
